internal/services/arcresourcebridge/arc_resource_bridge_appliance_data_source.go

```diff
@@ -0,0 +1,121 @@
+package arcresourcebridge
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+	"time"
+
+	"github.com/hashicorp/go-azure-helpers/lang/pointer"
+	"github.com/hashicorp/go-azure-helpers/lang/response"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/commonschema"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/identity"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/location"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/tags"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/resourceconnector/2022-10-27/appliances"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+)
+
+type ArcResourceBridgeApplianceDataSource struct{}
+
+var _ sdk.DataSource = ArcResourceBridgeApplianceDataSource{}
+
+func (r ArcResourceBridgeApplianceDataSource) ResourceType() string {
+	return "azurerm_arc_resource_bridge_appliance"
+}
+
+func (r ArcResourceBridgeApplianceDataSource) ModelObject() interface{} {
+	return &ApplianceModel{}
```

Comment:
We'll need a Data Source specific model here, and not re-use the resource model.